### PR TITLE
Upgrade Thanos to v0.34.1

### LINKED ISSUE #1903: [3] Upgrade Thanos to v0.32.5
**What Chart should be upgraded?**:

https://github.com/bitnami/charts/blob/main/bitnami/thanos/Chart.yaml

**Why should this Chart be upgraded?**:

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**:

- I checked the migration of the new Chart:
  - [ ] I upgraded a Chart and determined that no migration steps are needed.
  - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by lunkan93**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?
Upgrade Thanos to v0.34.1.

- Fixes #1903 

#### Information to reviewers
Some interesting changes:
- Store: breaking ⚠️ Changed --sync-block-duration default 3m to 15m.
  - Don't think this is an issue, let me know if you think otherwise.
- Query Frontend: breaking ⚠️ Add tenant label to relevant exported metrics. Note that this change may cause some pre-existing custom dashboard queries to be incorrect due to the added label.
  - Didn't find any issues caused by this
- Receive: add az-aware replication support for Ketama algorithm
  - Not sure if this will be interesting for us in the future but I thought I'd mention it here
- Default port for `queryFrontend` changed from `10902` to `9090` so I adjusted netpols
- Netpols were added upstream which I disabled.
- Had to fix values for receive-distributor to keep using the same SA

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [x] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [x] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [x] Any changed pod is covered by Pod Security Admission
  - [x] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [x] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [x] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests
- Config checks:
  - [ ] The schema was updated

### CODE REVIEW

File: helmfile.d/values/thanos/objectstorage-secret.yaml.gotmpl [hunk 1]
```diff
@@ -60,3 +79,23 @@ objstoreConfig: |-
...
+    enabled: false
+ruler:
+  networkPolicy:
+    enabled: false
+storegateway:
+  networkPolicy:
+    enabled: false
+query:
+  enabled: false
+  networkPolicy:
```
Comment (aarnq) [inline helmfile.d/values/thanos/objectstorage-secret.yaml.gotmpl:100]: Added twice?

File: helmfile.d/values/networkpolicies/service/grafana.yaml.gotmpl [hunk 2]
```diff
@@ -27,7 +27,7 @@ policies:
               podSelectorLabels:
                 app.kubernetes.io/component: query-frontend
           ports:
-            - tcp: 10902
+            - tcp: 9090
```
Comment (aarnq) [inline helmfile.d/values/networkpolicies/service/grafana.yaml.gotmpl:30]: So, is grafana and grafana-label-enforcer updated to use these ports? Or does it use port names?

Comment (lunkan93) [reply helmfile.d/values/networkpolicies/service/grafana.yaml.gotmpl:30]: I did not find that this was needed as they are targeting the service, e.g. https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile.d/values/grafana/grafana-label-enforcer.yaml.gotmpl#L3
Unless I'm missing something? :thinking:

Comment (lunkan93) [reply helmfile.d/values/networkpolicies/service/grafana.yaml.gotmpl:30]: (Since the containerPort is what has changed)

Comment (aarnq) [reply helmfile.d/values/networkpolicies/service/grafana.yaml.gotmpl:30]: Ah, strange :sweat_smile:

**Review by aarnq — APPROVED** [review]

**Review by robinAwallace — APPROVED** [review]

**Review by Xartos — APPROVED** [review]

**Pull request merged**